apps/calendar/src/components/hooks/drag.ts

```diff
@@ -20,86 +27,106 @@ export function useDrag({
   onClick = noop,
   onCancel = noop,
 }: DragListeners) {
-  let onMouseMove: MouseEventListener;
-  let onMouseUp: MouseEventListener;
-  let onKeydownESC: (e: KeyboardEvent) => void;
-
-  let distance = 0;
-  let isMoved = false;
-  let isDragStartFired = false;
-
-  const clearData = () => {
-    distance = 0;
-    isDragStartFired = false;
-    isMoved = false;
-  };
-  const toggleListeners = (on: boolean) => {
-    if (on) {
-      document.addEventListener('mousemove', onMouseMove);
-      document.addEventListener('mouseup', onMouseUp);
-      document.addEventListener('keyup', onKeydownESC);
-    } else {
-      document.removeEventListener('mousemove', onMouseMove);
-      document.removeEventListener('mouseup', onMouseUp);
-      document.removeEventListener('keydown', onKeydownESC);
-    }
-  };
-
-  const onMouseDown = (e: MouseEvent) => {
-    // only primary button can start drag. 0 is primary button value.
-    if (e.button !== 0) {
-      return;
-    }
-
-    clearData();
-    toggleListeners(true);
-  };
+  const [isStarted, setStarted] = useState(false);
+  const movedDistanceRef = useRef(0);
+  const isDraggingRef = useRef(false);
+
+  const onMouseMoveRef = useRef<MouseEventListener | null>(null);
+  const onMouseUpRef = useRef<MouseEventListener | null>(null);
+  const onKeyDownRef = useRef<KeyboardEventListener | null>(null);
+
+  const resetState = useCallback(() => {
+    movedDistanceRef.current = 0;
+    isDraggingRef.current = false;
+    setStarted(false);
+  }, []);
+
+  const onMouseDown = useCallback<MouseEventListener>(
+    (e) => {
+      if (!isLeftClick(e.button)) {
```

Comment:
```suggestion
({ button }) => {
  if (!isLeftClick(button)) {
```
이제 보니 바로 디스트럭쳐링해서 사용해도 될 것 같습니다!